src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -145,37 +230,95 @@ def cli_cosmosdb_create(cmd, client,
     backup_policy = None
-    if backup_interval is not None or backup_retention is not None:
+    if backup_policy_type is not None:
+        if backup_policy_type.lower() == 'periodic':
+            backup_policy = PeriodicModeBackupPolicy()
+            if backup_interval is not None or backup_retention is not None:
+                periodic_mode_properties = PeriodicModeProperties(
+                    backup_interval_in_minutes=backup_interval,
+                    backup_retention_interval_in_hours=backup_retention
+                )
+            backup_policy.periodic_mode_properties = periodic_mode_properties
+        elif backup_policy_type.lower() == 'continuous':
+            backup_policy = ContinuousModeBackupPolicy()
+        else:
+            raise CLIError('backup-policy-type argument is invalid.')
+    elif backup_interval is not None or backup_retention is not None:
         backup_policy = PeriodicModeBackupPolicy()
         periodic_mode_properties = PeriodicModeProperties(
             backup_interval_in_minutes=backup_interval,
             backup_retention_interval_in_hours=backup_retention
         )
         backup_policy.periodic_mode_properties = periodic_mode_properties
 
-    params = DatabaseAccountCreateUpdateParameters(
-        location=resource_group_location,
-        locations=locations,
-        tags=tags,
-        kind=kind,
-        consistency_policy=consistency_policy,
-        ip_rules=ip_range_filter,
-        is_virtual_network_filter_enabled=enable_virtual_network,
-        enable_automatic_failover=enable_automatic_failover,
-        capabilities=capabilities,
-        virtual_network_rules=virtual_network_rules,
-        enable_multiple_write_locations=enable_multiple_write_locations,
-        disable_key_based_metadata_write_access=disable_key_based_metadata_write_access,
-        key_vault_key_uri=key_uri,
-        public_network_access=public_network_access,
-        api_properties=api_properties,
-        enable_analytical_storage=enable_analytical_storage,
-        enable_free_tier=enable_free_tier,
-        network_acl_bypass=network_acl_bypass,
-        network_acl_bypass_resource_ids=network_acl_bypass_resource_ids,
-        backup_policy=backup_policy,
-        identity=system_assigned_identity,
-        default_identity=default_identity)
+    create_mode = 'Default'
+    if is_restore_request is not None:
+        create_mode = 'Restore' if is_restore_request else 'Default'
+
+    params = None
+    if create_mode == 'Restore':
+        if restore_source is None or restore_timestamp is None:
+            raise CLIError('restore-source and restore-timestamp should be provided for a restore request.')
+        restore_parameters = RestoreParameters(
+            restore_mode='PointInTime',
+            restore_source=restore_source,
+            restore_timestamp_in_utc=restore_timestamp
+        )
+        if databases_to_restore is not None:
+            logger.debug(databases_to_restore)
```

Comment:
Is this for debug during implementing? If you want to keep this log it will be better to add more info here or else pls remove this.